src/Symfony/Component/Security/Core/User/UserProviderInterface.php

```diff
@@ -39,8 +41,12 @@ interface UserProviderInterface
      * object can just be merged into some internal array of users / identity
      * map.
      *
+     * @param TUser $user
```

Comment:
technically, passing a different user is valid in the interface, as it defines that the implementation should throw a UnsupportedUserException in such case.